tests/test_responses.py

```diff
@@ -241,6 +242,31 @@ async def app(scope, receive, send):
     assert filled_by_bg_task == "6, 7, 8, 9"
 
 
+def test_file_response_with_range(tmpdir, test_client_factory):
+    path = os.path.join(tmpdir, "xyz")
+    content = b"<file content>"
+    with open(path, "wb") as file:
+        file.write(content)
+
+    async def app(scope, receive, send):
+        range_header = Headers(scope=scope)["range"]
+        start, end = (int(v) for v in range_header[len("bytes=") :].split("-"))
+        response = FileResponse(path=path, filename="example.png", range=(start, end))
```

Comment:
This is what a developer would need to do by its own with this solution.

It doesn't look good.

Reply:
What about something like... `FileResponse.with_range_from_headers(path=path, filename=..., headers=Headers(scope=scope))`?

Reply:
Hmm... No... But you actually gave me an idea:

```python
range_header = Headers(scope=scope)["range"]
FileResponse(path=path, filename=..., range_header=range_header)
```

Still... I'm not comfortable yet... :thinking: